crates/core/tedge_api/src/entity_store.rs

```diff
@@ -201,7 +204,9 @@ impl EntityStore {
             message_log,
         };
 
-        entity_store.load_from_message_log(log_dir.as_ref());
+        if !clean_start {
+            entity_store.load_from_message_log(log_dir.as_ref());
+        }
```

Comment:
As the log is wiped on a clean start, it's safe to load it unconditionally.

```suggestion
        entity_store.load_from_message_log(log_dir.as_ref());
```